Demo/MozillaSocial-iOS/MoSoContent/LoginView.swift

```diff
@@ -0,0 +1,24 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import SwiftUI
+
+struct LoginView: View {
+    let auth = Auth()
+
+    var body: some View {
+        Button("Login!") {
+            Task {
+                guard let clientEntity = await auth.registerApp() else {
+                    print("Client Entitiy Nil")
+                    return
+                }
+                auth.signIn(client: clientEntity)
```

Comment:
would it be possible to abstract the error handling logic away from `LoginView`? Ideally, we could have a `viewModel` that has an injected `Auth` instance, but in this simple case, I think it's fine to inject `Auth` directly into the view and perhaps add a method to it that handles the logic and such. If you want `LoginView` to be updated in any way after login succeeds/fails, you might want to make that an `ObservableObject`

Reply:
I didn't like that Auth stuff was happening in the view, so I moved that inside the Auth class (Which has also been renamed)